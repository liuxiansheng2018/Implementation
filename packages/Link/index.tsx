import {computed , getCurrentInstance} from 'vue'
import { createNameSpace,useRoute } from '../utils'

const [createComponent] = createNameSpace('Link')

export default createComponent({
    props: {
        href: {
            type: String,
            default: ''
        },
        to: {
            type: [String, Object],
            default: ''
        },
        color: Boolean,
        underline: Boolean,
        block: Boolean,
    },

    setup (props, {slots}) {
        const route = useRoute()
        const setClass = computed(()=> {
            const name: string[] = []
            props.color && name.push('color')
            props.underline && name.push('underline')
            props.block && name.push('block')
            return name.join(' ')
        })

        const safeHref = computed( ()=> {
            if( props.to ) return 'javascript: void 0;'
            return props.href
        })

        const goToHandler = () => props.to && route()


        return () => (
            <a class={`implementation-link ${setClass.value}`}
             href={safeHref.value}
             onClick={goToHandler}
            >
                {slots.default?.()}
            </a>
        )
    }
})