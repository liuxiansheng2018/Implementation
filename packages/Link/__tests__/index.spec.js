import { mount } from '@vue/test-utils'
import Link from '../index'

describe('Link', () => {
  it('should be render as element', () => {
    const link = mount(Link)
    expect(link.html()).toMatchSnapshot()
    expect(() => link.unmount()).not.toThrow()
  })

  it('should be render correctly', () => {
    const wrapper = mount({
      render() {
        return (
          <>
            <Link href="https://www.test.com">Link</Link>
            <Link href="https://www.test.com" color>
              Link
            </Link>
            <Link href="https://www.test.com" underline>
              Link
            </Link>
            <Link href="https://www.test.com" color underline>
              Link
            </Link>
            <Link href="https://www.test.com" block>
              Link
            </Link>
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
