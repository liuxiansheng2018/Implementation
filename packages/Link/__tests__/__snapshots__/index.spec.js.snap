// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should be render as element 1`] = `
<a class="implementation-link " href="">
  <!---->
</a>
`;

exports[`Link should be render correctly 1`] = `<a class="implementation-link " href="https://www.test.com">Link</a><a class="implementation-link color" href="https://www.test.com">Link</a><a class="implementation-link underline" href="https://www.test.com">Link</a><a class="implementation-link color underline" href="https://www.test.com">Link</a><a class="implementation-link block" href="https://www.test.com">Link</a>`;
